docs/docs/usage.md

```diff
@@ -872,6 +872,31 @@ tasks:
       - sleep 5 # long operation like installing packages
 ```
 
+### Ensuring required variables are set
+
+If you want to check that certain variables are set before running a task then
+you can use `requires`. This is useful when might not be clear to users which
+variables are needed, or if you want clear message about what is required. Also
+some tasks could have dangerous side effects if run with un-set variables.
+
+Using `requires` you specify a list of strings, these strings are variable names
+which are checked prior to running the task. If any variables are un-set the the
+task will error and not run. Environmental variables are also checked.
+
+> Note. Variables set to empty zero length strings, will pass the `requires` check
```

Comment:
Using the proper Docusaurus syntax for this:

```suggestion
:::note

Variables set to empty zero length strings, will pass the `requires` check.

:::
```

Reply:
Done